src/test/Makefile

```diff
@@ -265,15 +269,24 @@ telemetry_ibus_unittest_SRC := \
 		$(USER_DIR)/telemetry/ibus_shared.c \
 		$(USER_DIR)/telemetry/ibus.c
 
+timer_definition_unittest_EXPAND := yes
 
-transponder_ir_unittest_SRC := \
-	        $(USER_DIR)/drivers/transponder_ir_ilap.c \
-	        $(USER_DIR)/drivers/transponder_ir_arcitimer.c
+# SITL is a simulator; NERO and STM32F7X2 don't define USED_TIMERS in target.h.
+timer_definition_unittest_BLACKLIST := STM32F7X2 SITL NERO
```

Comment:
Instead of ignoring these targets, they should be fixed to define `USED_TIMERS`. In general, exclusions like in the case of SITL should be handled in the test code, and adding a dedicated stanza for exclusions avoided, since this makes the tests less self-contained.

Reply:
The three targets are blacklisted from the new test because they don't compile with the test code at all.

Of course I tried to introduce a `USED_TIMERS` definition there, but these targets seem to have a different structure of `#include` dependencies than the rest, so I got redefinitions of `USED_TIMERS`, conflicting with a "generic" version defined elsewhere. So I gave up on adding `USED_TIMERS`.

If you want me to introduce `#define USED_TIMERS` to these targets, I'll give it another try. I'll need to investigate why the duplicate definition issue affects only these targets.

Another option is to only check for `timerHardware` consistency in these 3 targets and to `#ifdef` away the test methods that touch `USED_TIMERS`.

Reply:
Ah yes, now I see. NERO was converted to dynamic timer management, and STM32F7X2 uses this as well, as it is a universal target. The plan is that for the 4.0 release all targets will be converted to this. Considering this, I don't think there is much that can be done to keep the test for `USED_TIMERS` around for much time, since all timers will be dynamic. Not sure if anything can be done to make it do something of value for targets with dynamic timer management, since the evaluation of which timers are used is done at runtime.

Reply:
After some investigation, #6760 has a partial fix.

1. Turns out I had been breaking `NERO` while trying to fix `SITL`. `NERO` appears to work on its own.
2. The symbol conflict was because of `USE_TIMER_MGMT`; it should build fine now.
3. `SITL` is, IMHO, still a no-go for coverage with this test.

Reply:
> Ah yes, now I see. NERO was converted to dynamic timer management, and STM32F7X2 uses this as well, as it is a universal target. The plan is that for the 4.0 release all targets will be converted to this. Considering this, I don't think there is much that can be done to keep the test for `USED_TIMERS` around for much time, since all timers will be dynamic. Not sure if anything can be done to make it do something of value for targets with dynamic timer management, since the evaluation of which timers are used is done at runtime.

Ah! OK. Then let me amend #6760 accordingly. What I did there is wrong then.